src/foraging_gui/Foraging.py

```diff
@@ -2299,30 +2303,48 @@ def _GiveRight(self):
         self.ManualWaterVolume[1]=self.ManualWaterVolume[1]+float(self.TP_GiveWaterR_volume)/1000
         self._UpdateSuggestedWater()
 
+
     def _UpdateSuggestedWater(self,ManualWater=0):
         '''Update the suggested water from the manually give water'''
-        Tag=0
-        if self.TotalWater.text()!='':
-            # self.BS_TotalReward: normal rewards and auto rewards
+        try:
+            if self.BaseWeight.text()!='' and self.TargetRatio.text()!='':
+                # set the target weight
+                target_weight=float(self.TargetRatio.text())*float(self.BaseWeight.text())
+                self.TargetWeight.setText(str(np.round(target_weight,3)))
+
             if hasattr(self,'GeneratedTrials'):
                 if hasattr(self.GeneratedTrials,'BS_TotalReward'):
-                    self.B_SuggestedWater=float(self.TotalWater.text())-float(self.GeneratedTrials.BS_TotalReward)/1000-np.sum(self.ManualWaterVolume)
-                    self.SuggestedWater.setText(str(np.round(self.B_SuggestedWater,3)))
+                    BS_TotalReward=float(self.GeneratedTrials.BS_TotalReward)/1000
                 else:
-                    Tag=1
+                    BS_TotalReward=0
             else:
-                Tag=1
-            if Tag==1:
-                if self.SuggestedWater.text()=='':
-                    try:
-                        self.SuggestedWater.setText(self.TotalWater.text())
-                    except Exception as e:
-                        logging.error(str(e))
-                try:
-                    self.B_SuggestedWater=float(self.TotalWater.text())-np.sum(self.ManualWaterVolume)
-                    self.SuggestedWater.setText(str(np.round(self.B_SuggestedWater,3)))
-                except Exception as e:
-                    logging.error(str(e))
+                BS_TotalReward=0
+
+            if hasattr(self,'ManualWaterVolume'):
+                ManualWaterVolume=np.sum(self.ManualWaterVolume)
+            else:
+                ManualWaterVolume=0
+            Earnedwater=BS_TotalReward+ManualWaterVolume
+            if self.WeightAfter.text()!='' and self.BaseWeight.text()!='' and self.TargetRatio.text()!='':
+                # calculate the suggested water
+                suggested_water=target_weight-float(self.WeightAfter.text())
+                # give at lease 1ml
+                if suggested_water<1:
+                    suggested_water=1-Earnedwater
+                if suggested_water<0:
+                    suggested_water=0
```

Comment:
> To be more concise, the whole block could be replaced by `suggested_water = max(0, target_weight - float(self.WeightAfter.text(), 1 - Earnedwater)` See our discussion today [AllenNeuralDynamics/aind-behavior-blog#111](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/111)

On both conditions, the minimum total water (supplemental water and the water consumed during that session) is 1ml. The difference is when to check we should use "post weight-base weight" or "1-earned water". My criteria is to check the post weight-base weight>1. The criteria proposed in https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/111 is to check post weight-base weight>1-earned water. I thought we misunderstood Jackie's explanation in Tuesday's meeting. My understanding is we should use " post weight-base weight>1

Reply:
We agree there are two steps.
1) Follow exactly what Mindscope is doing.
2) Are these standards reasonable?

Reply:
> The criteria proposed in https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/111 will always give more water to match the target weight.

I don't agree with this. It will not **always** give more water to match the target weight. It gives more water than what needed for the target weight **only when** the second constraint suggests more water, i.e., `1 - earned > target - post`, which is exactly what we want under the Mindscope algorithm.

Reply:
Yes, it will only give more water when target-post<1-earned. I do not know the rationale.

Reply:
The rationale is we can simply write these two lines in IACUC:
> 1. When we put the animal back to the vivarium, the weight is ***at least*** `target_weight`.
> 2. Each animal receives ***at least*** 1.0 ml total water per day.

So the rationale is clear to me. I agree that this may not be the best algorithm because we haven't put any behavioral metrics into the equation, and we can further optimize the two hyperparameters ("1.0 ml" and "target_weight"), but these are separate discussions.

On the other hand, I'm not clear about the rationale behind your code:
> In short, my criteria will give less water ((target weight-post weight)-(1-earned)) when 1-earned<target-post weight<1.

This effectively adds another condition to the two conditions above, together with another hyperparameter (for example, you can do `give less water ((target weight-post weight)-(1-earned)) when 1-earned < target-post weight < 2`). According to Occam's razor, you should actually provide more evidence to defend the rationale.

Reply:
I agree with you that the standards ttps://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/111 are more simplified. However, a simplified definition does not mean it works better. I just hope to know how did the method used in the mindscope keeps the body weight and motivation. Again, I don't prefer either one of them. There is high probability both of them fail and work given the difference is occurred only when **1-earned<target-post weight<1**.

Reply:
I checked the mindscope GUI and confirmed that they use standards https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/111. I will make changes. When we have collected enough data, we should evaluate its effectiveness.